test/common_utils.py

```diff
@@ -1111,15 +1111,42 @@ def random_fullrank_matrix_distinct_singular_value(matrix_size, *batch_dims,
     dtype = kwargs.get('dtype', torch.double)
     device = kwargs.get('device', 'cpu')
     silent = kwargs.get("silent", False)
+    singular = kwargs.get("singular", False)
     if silent and not torch._C.has_lapack:
         return torch.ones(matrix_size, matrix_size, dtype=dtype, device=device)
 
     A = torch.randn(batch_dims + (matrix_size, matrix_size), dtype=dtype, device=device)
     u, _, v = A.svd()
     s = torch.arange(1., matrix_size + 1, dtype=dtype, device=device).mul_(1.0 / (matrix_size + 1)).diag()
+    if singular:
+        s[-1, -1] = 0
+        if matrix_size > 2:
+            s[0, 0] = 0
     return u.matmul(s.expand(batch_dims + (matrix_size, matrix_size)).matmul(v.transpose(-2, -1)))
 
 
+def random_matrix_distinct_singular_value(rows, columns, *batch_dims,
+                                          **kwargs):
+    dtype = kwargs.get('dtype', torch.double)
+    device = kwargs.get('device', 'cpu')
+    silent = kwargs.get("silent", False)
+    singular = kwargs.get("singular", False)
+    if silent and not torch._C.has_lapack:
+        return torch.ones(rows, columns, dtype=dtype, device=device)
+
+    A = torch.randn(batch_dims + (rows, columns), dtype=dtype, device=device)
+    u, _, v = A.svd(some=False)
+    s = torch.zeros(rows, columns, dtype=dtype, device=device)
+    k = min(rows, columns)
+    for i in range(k):
+        s[i, i] = (i + 1) / (k + 1)
+    if singular:
+        s[k - 1, k - 1] = 0
+        if k > 2:
+            s[0, 0] = 0
```

Comment:
I didn't quite understand the rationale behind this change.

Reply:
This is to make sure that there will be non-trivial pivots for the tests.

Reply:
Added comments to te code.